libvast/src/arrow_table_slice.cpp

```diff
@@ -611,29 +611,37 @@ const record_type& arrow_table_slice<FlatBuffer>::layout() const noexcept {
 
 template <class FlatBuffer>
 table_slice::size_type arrow_table_slice<FlatBuffer>::rows() const noexcept {
-  return record_batch()->num_rows();
+  if (auto&& batch = record_batch())
+    return batch->num_rows();
+  return {};
 }
 
 template <class FlatBuffer>
 table_slice::size_type arrow_table_slice<FlatBuffer>::columns() const noexcept {
-  return record_batch()->num_columns();
+  if (auto&& batch = record_batch())
+    return batch->num_columns();
+  return {};
```

Comment:
```suggestion
  return 0;
```

I think we can assume that there exists an implicit conversion from 0 to the `size_type`. This is more idiomatic.

Reply:
I don't mind either way.